flink-table/flink-table-planner-blink/src/main/scala/org/apache/flink/table/planner/plan/utils/RexNodeExtractor.scala

```diff
@@ -256,7 +256,15 @@ class RefFieldAccessorVisitor(usedFields: Array[Int]) extends RexVisitorImpl[Uni
     if (right.length < left.length) {
       false
     } else {
-      right.take(left.length).equals(left)
+      right.take(left.length).zip(left).foldLeft(true) {
+        (ans, fields) => {
+          if (ans) {
+            fields._1.equals(fields._2)
```

Comment:
It's better we does not use `_1`, `_2`, which makes the code hard to read. we can use use `case` match with meaningful name, just like:
```
right.take(left.length).zip(left).foldLeft(true) {
       case (ans, (rName, lName)) => {
          if (ans) {
            lName.equals(rName)
          } else {
            false
          }
        }
      }
```